lib/dotcom_web/live/trip_planner.ex

```diff
@@ -67,22 +71,31 @@ defmodule DotcomWeb.Live.TripPlanner do
             <.itinerary_group :for={group <- groups} group={group} />
           </div>
         </.async_result>
-        <div id="trip-planner-map-wrapper" class="w-full" phx-update="ignore">
-          <div style="min-height: 400px;" id="trip-planner-map" phx-hook="TripPlannerMap" />
-        </div>
+        <.live_component
+          module={MbtaMetro.Live.Map}
+          id="trip-planner-map"
+          class="h-96 w-full relative overflow-none"
+          config={@map_config}
+          pins={@pins}
```

Comment:
```suggestion
          pins={[@from, @to]}
```

This, along with changing `update_pin_in_socket` to simply:

```elixir
defp update_pin_in_socket(socket, [longitude, latitude], direction)
     when longitude != "" and latitude != "" do
  socket
  |> assign(direction, [String.to_float(longitude), String.to_float(latitude)])
end

defp update_pin_in_socket(socket, [longitude, latitude], direction)
     when longitude == "" or latitude == "" do
  socket
  |> assign(direction, [])
end
```

And adding
```elixir
|> assign(:from, [])
|> assign(:to, [])
```
To the socket initialization in `mount`...

Would allow removing all of the `place_pin` logic, and most of the more complicated guts of the `update_pin_in_socket` functions as well.

(Net reduction of 41 lines of code 😇)

I _think_ it would also be possible to combine `update_pin_in_socket` with `update_*_pin`, potentially for an even greater LoC reduction (and maybe clearer code), but this feels like a start.

Reply:
Oooh actually, yes. If we replace all of the pin stuff with

```elixir
 defp update_from_pin(socket, %{"from" => from}) do
   assign(socket, :from, to_geojson(from))
 end

 defp update_to_pin(socket, %{"to" => to}) do
   assign(socket, :to, to_geojson(to))
 end

 defp to_geojson(%{"longitude" => longitude, "latitude" => latitude})
      when longitude != "" and latitude != "" do
   [String.to_float(longitude), String.to_float(latitude)]
 end

 defp to_geojson(_coordinates) do
   []
 end
```

That gets us a net reduction of 53 lines of code (-54, but +1 because we have to set two `assigns` in the `mount`), and it resolves my concern with abruptly changing to geojson format in the middle of the live view, since this way we can keep the geojson format closer to the component where it's actually being used.

Reply:
had to add in another 6 lines of code because you have to have defaults for update_from_pin and update_to_pin for when the pattern doesn't match.